instrumentation/opentelemetry-instrumentation-celery/src/opentelemetry/instrumentation/celery/utils.py

```diff
@@ -95,8 +95,6 @@ def set_attributes_from_context(span, context):
                     SpanAttributes.MESSAGING_DESTINATION, routing_key
                 )
 
-            value = str(value)
```

Comment:
I think this does match with the value an attributes can have https://opentelemetry.io/docs/concepts/signals/traces/#attributes

Reply:
Yes, I opened [this](https://github.com/open-telemetry/opentelemetry-python/issues/4380) and realized that it is intentional.

Right now, since we do `value = str(value)` the map becomes tricky to parse (filtering based on key).

How about if I flatten the map and then stringily (or convert to corresponding datatype) the values?
```python
celery.delivery_info ⇒ `{'exchange': '', 'routing_key': 'distilled-guidance-dpo-25-10k-v1', 'priority': 0, 'redelivered': False}`
```
would become
```python
celery.delivery_info.exchange: "routing_key"
celery.delivery_info.routing_key: "distilled-guidance-dpo-25-10k-v1"
celery.delivery_info.priority: 0
celery.delivery_info.redelivered: false
```

Reply:
Flattening the map would indeed be better I think than going out of spec :)

Reply:
I have updated the changes to add span attributes with flattened map. PTAL, thanks!